addons/account/models/account_move.py

```diff
@@ -3715,14 +3715,16 @@ def _check_constrains_account_id_journal_id(self):
 
     @api.constrains('account_id', 'tax_ids', 'tax_line_id', 'reconciled')
     def _check_off_balance(self):
-        for line in self:
-            if line.account_id.internal_group == 'off_balance':
-                if any(a.internal_group != line.account_id.internal_group for a in line.move_id.line_ids.account_id):
-                    raise UserError(_('If you want to use "Off-Balance Sheet" accounts, all the accounts of the journal entry must be of this type'))
-                if line.tax_ids or line.tax_line_id:
-                    raise UserError(_('You cannot use taxes on lines with an Off-Balance account'))
-                if line.reconciled:
-                    raise UserError(_('Lines from "Off-Balance Sheet" accounts cannot be reconciled'))
+        off_balance_line = self.search([('id', 'in', self.ids), ('account_id.internal_group', '=', 'off_balance'), ])
+        non_off_balance_line = self.search([('id', 'in', self.ids), ('account_id.internal_group', '!=', 'off_balance'), ])
+        balance_in_off_balance = sum(l.debit - l.credit for l in off_balance_line)
+        balance_in_non_off_balance = sum(l.debit - l.credit for l in non_off_balance_line)
```

Comment:
This could be obtained in a more pythonic/odoo way with a sum of a mapped of a filtered:

```suggestion
        off_balance_lines = self.filtered(lambda r: r.account_id.internal_group == 'off_balance')
        non_off_balance_lines = self.filtered(lambda r: r.account_id.internal_group != 'off_balance')
        balance_in_off_balance = sum(off_balance_lines.mapped(lambda l: l.debit - l.credit))
        balance_in_non_off_balance = sum(non_off_balance_lines.mapped(lambda l: l.debit - l.credit))
```

Reply:
Check the new approach